### DISCUSSION TITLE
Dynamic import callback missing error after moving from webpack v4 to v5

### DISCUSSION THREAD

**Discussion by rushill2** [webpack/webpack#18362] [Q&A]
Hello all, I am on node `20.11.0` and just moved to webpack 5.91.0, and while building my production build I always receive the error

`ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING`

The changes I have made to my babel configs for the base, dev and prod environments are as follows:

1. devtool: 'cheap-module-eval-source-map', was changed to   devtool: 'eval-cheap-module-source-map',
2. Removed `new webpack.NamedModulesPlugin(),`
3. Changed `node: { fs: 'empty' },` to ```
`node: {
    global: false,
    __dirname: false,
    __filename: false,
  },
```
4. And directly used the string replace loader instead of using the plugin.


My local environment runs perfectly, but the production build always fails with this issue.

```
node:internal/modules/esm/utils:213
  throw new ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING();
        ^

TypeError: A dynamic import callback was not specified.

```
Seems to be a node issue but could anyone point me in the right direction please?I can provide further details as needed, but is there anything that I have missed directly from these details? Appreciate the help!

**Comment by alexander-akait** [comment]
Please create reproducible test repo, thank you
